plugins/nodes/src/js/pages/nodes-overview/HostsPageContent.js

```diff
@@ -0,0 +1,123 @@
+import PureRenderMixin from 'react-addons-pure-render-mixin';
+import React from 'react';
+import {RouteHandler} from 'react-router';
+
+import FilterBar from '../../../../../../src/js/components/FilterBar';
+import FilterButtons from '../../../../../../src/js/components/FilterButtons';
+import FilterByService from '../../../../../services/src/js/components/FilterByService';
+import FilterHeadline from '../../../../../../src/js/components/FilterHeadline';
+import ResourceBarChart from '../../../../../../src/js/components/charts/ResourceBarChart';
+
+const HEALTH_FILTER_BUTTONS = ['all', 'healthy', 'unhealthy'];
+
+const METHODS_TO_BIND = ['onResetFilter'];
+
+class HostsPageContent extends React.Component {
+
+  constructor() {
+    super(...arguments);
+    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate;
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  onResetFilter() {
+    this.props.onResetFilter(...arguments);
+
+    if (this.serviceFilter !== null && this.serviceFilter.dropdown !== null) {
+      this.serviceFilter.setDropdownValue('default');
```

Comment:
We could also think of a pattern for these that get set from refs. It took me a minute to figure out where this was being set as it wasn't obvious at first.